### PR TITLE
SIGN-6714 - unzip the signed artifact ZIP file downloaded from SignPath

### PR DESCRIPTION
**Pull request by volbobvol**
When the submit-signing-request action downloads the signed artifact from SignPath, it should unzip the content and place it in the location specified by the 'output-artifact-directory' field.

### CODE REVIEW

File: actions/submit-signing-request/task.ts [hunk 1]
```diff
@@ -231,32 +232,35 @@ export class Task {
...
+        const zip = new nodeStreamZip.async({ file: tmpZipFile });
+        await zip.extract(null, targetDirectory);
 
-    async logArtifactFileStat(artifactPath: string) {
-        await fs.stat(artifactPath, (err, stats) => {
-            const size = fileSize.partial({base: 2, standard: "jedec"});
-            core.info("File size: " + size(stats.size));
-        });
+        // delete temp ZIP file
+        fs.unlinkSync(tmpZipFile);
```
Comment (paulsavoie) [inline actions/submit-signing-request/task.ts]: Should be cleaned up even if an error occurs above, imo

File: actions/submit-signing-request/task.ts [hunk 2]
```diff
@@ -231,32 +232,35 @@ export class Task {
...
-        if (!fs.existsSync(targetDir)) {
-            core.info(`Directory ${targetDir} does not exist, and will be created`);
-            fs.mkdirSync(targetDir, { recursive: true });
-        }
+        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetDirectory}`);
 
-        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetFilePath}`);
-        const writer = fs.createWriteStream(targetFilePath)
+        // save the signed artifact to temp ZIP file
+        const tmpZipFile = path.join(targetDirectory, '__signpath_signed_artifact_tmp.zip');
```
Comment (paulsavoie) [inline actions/submit-signing-request/task.ts]: Please create a temporary directory (e.g. using https://nodejs.org/api/fs.html#fspromisesmkdtempprefix-options) instead of storing the artifact with a fixed name next to the extracted content.

I was hoping that we could just pass the HttpStream from axios to the zipping library instead of storing the zip file first, but looking at the APIs doesn't seem to be possible.

File: actions/submit-signing-request/task.ts [hunk 3]
```diff
@@ -231,32 +232,35 @@ export class Task {
...
+        const writer = fs.createWriteStream(tmpZipFile)
```
Comment (paulsavoie) [inline actions/submit-signing-request/task.ts]: Not necessary, but since you're consistent otherwise: colon (;) missing at the end ;-)

File: actions/submit-signing-request/task.ts [hunk 4]
```diff
@@ -231,32 +232,35 @@ export class Task {
...
         response.data.pipe(writer);
         await new Promise((resolve, reject) => {
-            writer.on('finish', resolve)
-            writer.on('error', reject)
+             writer.on('finish', resolve)
+             writer.on('error', reject)
```
Comment (paulsavoie) [inline actions/submit-signing-request/task.ts]: Why did you add these extra spaces for intentation here?

File: actions/submit-signing-request/task.ts [hunk 5]
```diff
@@ -231,32 +232,35 @@ export class Task {
...
+        // delete temp ZIP file
+        fs.unlinkSync(tmpZipFile);
+
+        core.info(`The signed artifact has been successfully downloaded from SignPath and extracted to ${targetDirectory}`);
     }
 
+    resolveOrCreateDirectory(relativePath:string): string {
+        const absolutePath = path.join(process.env.GITHUB_WORKSPACE as string, relativePath)
+        if (!fs.existsSync(absolutePath)) {
+            core.info(`Directory ${absolutePath} does not exist, and will be created`);
```
Comment (paulsavoie) [inline actions/submit-signing-request/task.ts]: Without the comma, so it should be "... does not exist and will be created"

Comment (paulsavoie) [reply actions/submit-signing-request/task.ts]: Also, should we put values in quotes? like Directory `"${absolutePath}"` - same for the other logs?

File: actions/submit-signing-request/task.ts (hunk 1 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: added finally block

File: actions/submit-signing-request/task.ts (hunk 2 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: Done

File: actions/submit-signing-request/task.ts (hunk 3 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: Fixed

File: actions/submit-signing-request/task.ts (hunk 4 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: Removed.

File: actions/submit-signing-request/task.ts (hunk 5 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: fixed

File: actions/submit-signing-request/task.ts [hunk 6]
```diff
@@ -264,8 +264,8 @@ export class Task {
         finally {
 
             core.debug(`Deleting temp directory ${tmpDir}`);
-            // delete the temp DIR
-            fs.unlinkSync(tmpDir);
+            // delete the temp ZIP
+            fs.unlinkSync(tmpZipFile);
```
Comment (paulsavoie) [inline actions/submit-signing-request/task.ts]: Yes, it should work to remove the entire directory. Just not with `unlinkSync`, but with `rmSync`.

Background:
https://www.geeksforgeeks.org/node-js-fs-unlinksync-method/
> This function does not work on directories, therefore it is recommended to use fs.rmdir() to remove a directory.

However, to remove recursively (including files), https://nodejs.org/api/fs.html#fsrmdirsyncpath-options says
> To get a behavior similar to the `rm -rf` Unix command, use fs.rmSync() with options { recursive: true, force: true }.

Please let's try that to properly clean up

**Review by paulsavoie — CHANGES_REQUESTED** [review]

File: actions/submit-signing-request/task.ts (hunk 6 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: As discussed, I use the runner TMP directory (RUNNER_TEMP) and do not remove the temporary file. According to GitHub Actions documentation, this folder is cleaned after each job run.

File: actions/submit-signing-request/task.ts (hunk 1 above)
Comment (volbobvol) [reply actions/submit-signing-request/task.ts]: eventually removed this block of code. Now rely on the agent and let it remove our temp files
